### ISSUE TITLE
Playout ended by itself after 4 days of playback

### ISSUE THREAD

**Issue by wizquest-labs** [ffplayout/ffplayout#140]
**Describe the bug**
I started the playout 4 days back, with the parameters

`nohup ffplayout -p ~/config.json -m playlist -o hls -i > nohup.out &`

config.json consists of 6 hours of video playback videos with remote source.

My expectation was that for 6 hours, it will play the files from remote source, and for the rest of the day, play default black frame.

Just before the start of the next day, I would update the date in config.json. For 4 days, it loaded the updated playlist fine with the message : "validate playlist from: <date>".

Today when I checked it, after the update of playlist, the program crashed with the exception: 

`thread 'main' panicked at 'index out of bounds: the len is 400 but the index is 478', src/input/playlist.rs:280:32`

I did not change the playlist data, except the date.

**Expected behavior**
The playout should continue for every date change

**Desktop/Server/Software (please complete the following information):**
 - OS: [e.g. debian 11]  Ubuntu 20.04.4 LTS
- ffplayout version : 0.9.8
- ffmpeg version: ffmpeg version 4.2.7-0ubuntu0.1 
- are you using the current master of ffplayout?

**Config Settings:**
- command line arguments : `nohup ffplayout -p ~/config.json -m playlist -o hls -i > nohup.out &`
- config fie
[config.json.txt](https://github.com/ffplayout/ffplayout/files/8976649/config.json.txt)

**Logging:**
- content of: ffplayout.log
[playout_logs.log](https://github.com/ffplayout/ffplayout/files/8976664/playout_logs.log)

**Comment by jb-alvarado** [comment]
Thank you for report, I will try to reproduce this error and fix it.

Just some notes before:
1. Changing the date in playlist will have not effect, except it force the playout to reload the playlist. When you give the playlist as a argument the playout will always take this playlist.
2. it would be better when you name the playlist other then config. Because the config would be the file /etc/ffplayout.yml. It brings a bit confusion.
3. can you post the content of the ffplayout.yml? At least the part from `playlist:`
4. I would not recommend to use the HLS output mode, but stream to a rtmp target. This will not produce a continous stream
5. when the most time you streaming just black, would be not better to use a cron job, with a fixed starting time and the let the stream end, when the playlist is done?

**Comment by jb-alvarado** [comment]
Should be fixed now in master. The next days I will make a release out of it.

But I had a another issue to: I had not finished integrate the option **infinite**. Know I add that, but when you run now with `--infinite` parameter, you don't get a black picture when your playlist is not long enough, instead the playlist content will repeat infinitely. If you need the black filler, remove the `-i` parameter.

**Comment by wizquest-labs** [comment]
@jb-alvarado thanks for the prompt support. I'll test the release and update you.

As a reply to your notes:

1. I was trying to reload the playlist, to just test the continuity of the stream. I'll be playing around with some configurations to explore and understand better, what are the various options which can help us create a world-class playout on cloud.
2. I agree with the naming convention. I'll update it, so that its understandable.
3. I'll setup again and update you on the contents, after testing on the new version.
4. Thanks for that heads-up. Will keep that in mind, during my next test
5. I agree. But I wanted to test the end-to-end delivery from playout, to transcoding, to delivery via CDN. That's why I just added few contents, and let the stream run continuously to test the different parts of the system.

Related to `"infinite"` parameter, I could understand that the Rust version is yet to be working. Ideally, we would want the same contents to be repeated if the user has not put new contents. Since `-i` was not working as expected, I just kept it there so that atleast the configuration is sorted. Correct stream generation can always be corrected when the `"infinte"` option is added.
